processor/src/main/java/org/mapstruct/ap/internal/model/assignment/SetterWrapperForCollectionsAndMaps.java

```diff
@@ -65,17 +63,35 @@ public Assignment getNewCollectionOrMapAssignment() {
     }
 
     @Override
+    public List<Type> getThrownTypes() {
+        List<Type> parentThrownTypes = super.getThrownTypes();
+        List<Type> result = new ArrayList<Type>( parentThrownTypes );
+        for ( Type thrownTypeToExclude : thrownTypesToExclude ) {
+            for ( Type parentThrownType : parentThrownTypes ) {
+                if ( parentThrownType.isAssignableTo( thrownTypeToExclude ) ) {
```

Comment:
Is this one related to the actual issue? And shouldn't this logic be applied in a more central location than only this particular assignment?